contracts/lib/BoundsLibrary.sol

```diff
@@ -0,0 +1,40 @@
+/*
+    Copyright 2019 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+pragma experimental "ABIEncoderV2";
+
+
+library BoundsLibrary {
+    /* ============ Structs ============ */
+    struct Bounds {
+        uint256 min;
+        uint256 max;
+    }
+
+    /* ============ Functions ============ */
+    function isValid(Bounds memory _bounds) internal pure returns(bool) {
```

Comment:
Interesting that this one is memory while the other ones are storage. I wonder if we should have both versions for every function

Reply:
We can call the storage one `isWithinS` or `isWithinStorage`

Reply:
Thinking about it, would it work for these to be memory? I think that when a storage element is passed in as a parameter, its automatically turned to memory.

Also, libraries in general, should aim to be in memory so it can be used by a larger variety of purposes